helpers

```diff
@@ -54,15 +54,14 @@ update_repo() {
 		git pull origin master
 	fi
 	set +e
-	# checkout the tag relating to the clear version used to build against
-	git checkout tags/"$CLRVER"
-	local branch="${CLRVER}_mix"
-	git rev-parse --verify "$branch"
+	# Ensure that CLRVER exists and is a tag
+	git rev-parse --verify "${CLRVER}^{tag}" &> /dev/null
 	if [ $? -eq 0 ]; then
-		git checkout "$branch"
-		git pull
-		else
-		git checkout -b "$branch"
+		# Always reset the _mix branch to be based off the tag
+		git checkout -B "${CLRVER}_mix" "${CLRVER}"
```

Comment:
The -B option doesn't fail if the branch already exists. I agree that a `git clean` might be helpful.